### PR TITLE
Github Runner: test use of self hosted runner for build

### PR DESCRIPTION
**Pull request by apentori**
**Work in progress**

Branch to test the deployment of self hosted linux github runner

### CODE REVIEW

**Review by jakubgs — APPROVED** [review]

**Comment by apentori** [comment]
The runner are hosted on Ubuntu 22.04 server

**Review by etan-status — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
         include:
           - target:
               os: linux
-            builder: ubuntu-20.04
+            builder: [self-hosted]
```
Comment (arnetheduck) [inline .github/workflows/ci.yml]: let's maybe keep both around for a bit until things are stable

**Comment by arnetheduck** [comment]
> The runner are hosted on Ubuntu 22.04 server

per discussion, we should use the oldest supported ubuntu LTS release here, meaning 20.04 in this case

**Comment by apentori** [comment]
The runner are being reinstalled on ubuntu 20.04.

First test detected a problem in step `Build Nim and Nimbus dependencies` at getting git submodudule for `vendor/nimbus-build-system/vendor/Nim`.

Currently investigating

File: .github/workflows/ci.yml (hunk 1 above)
Comment (apentori) [reply .github/workflows/ci.yml]: Tag `ubuntu-20.04` added in the list of builder and as a label for the runner

**Comment by apentori** [comment]
Runner updated to ubuntu 22.04

**Review by jakubgs — APPROVED** [review]

**Comment by jakubgs** [comment]
Seems like we have some builds already running:

image

I thought that maybe I could find some indication what host it's running on, but I can't find anything for this nightly:
https://github.com/status-im/nimbus-eth2/actions/runs/6491357946/job/17628544161

But the runners are definitely there:

image

**Pull request merged**